core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/UpgradeService.java

```diff
@@ -74,18 +75,18 @@
     @Inject
     private TransactionService transactionService;
 
-    public UpgradeOptionV4Response getUpgradeOptionByStackName(Long workspaceId, String stackName, User user) {
+    public UpgradeOptionV4Response getUpgradeOptionByStackNameOrCrn(Long workspaceId, NameOrCrn nameOrCrn, User user) {
         try {
             return transactionService.required(() -> {
-                Optional<Stack> stack = stackService.findStackByNameAndWorkspaceId(stackName, workspaceId);
+                Optional<Stack> stack = stackService.findStackByNameOrCrnAndWorkspaceId(nameOrCrn, workspaceId);
                 if (stack.isPresent()) {
                     try {
                         return getUpgradeOption(stack.get(), workspaceId, user);
                     } catch (CloudbreakImageNotFoundException | CloudbreakImageCatalogException e) {
                         throw new BadRequestException(e.getMessage());
                     }
                 } else {
-                    throw notFoundException("Stack", stackName);
+                    throw notFoundException("Stack", nameOrCrn.toString());
```

Comment:
small thing: why are you using toString here instead of name?

Reply:
Yes, that does not make sense any more. In the first implementation if you called getName() but it had the crn instead, there was no exception thrown. So I just wanted to make sure that it is logged to see the class in full.
In the present implementation if you call the wrong get method (name instead of crn) you get an exception and will be notified of the mismatch => so toString here does not make much sense any more.

Reply:
Long story short, I will change it.

Reply:
On this very place I need to use toString as nameOrCrn can hold either name or crn, it is not known statically. NameOrCrn object is passed to the method findStackByNameOrCrnAndWorkspaceId accepting both name and crn.

Reply:
@horadla23 Tests: I ran e2e aws tests locally:
testCreateScaleStopStartCluster
testSDXRepairIDBRokerWithStoppedEC2Instance